src/java/main/org/apache/zookeeper/server/persistence/FileTxnLog.java

```diff
@@ -199,14 +199,16 @@ public synchronized boolean append(TxnHeader hdr, Record txn)
                 LOG.warn("Current zxid " + hdr.getZxid()
```

Comment:
I think:
```
if (hdr == null)  return false;
/**/
return true; 
```
may be better,to avoid too deep if-else nest
BTW,why do I see so many flaky changes?